nodeup/pkg/model/context.go

```diff
@@ -485,12 +485,13 @@ func (c *NodeupModelContext) BuildPrivateKeyTask(ctx *fi.ModelBuilderContext, na
 func (c *NodeupModelContext) NodeName() (string, error) {
 	// This mirrors nodeutil.GetHostName
 	hostnameOverride := c.Cluster.Spec.Kubelet.HostnameOverride
```

Comment:
I don't think you need to recognize the user-set hostname. As far as I can tell, you cannot set this to anything meaningful. You can just add documentation to the API field (`pkg/apis/kops`) that setting this flag is deprecated and is ignored as of kOps 1.20.

Then you can test below only based on `cloudProvider`, which should make things a lot simpler for you.